src/actix/certificate_helpers.rs

```diff
@@ -109,21 +111,17 @@ fn load_certified_key(tls_config: &TlsConfig) -> io::Result<Arc<CertifiedKey>> {
         })
         .collect();
     if certs.is_empty() {
-        return Err(io::Error::new(
-            io::ErrorKind::Other,
-            "No server certificate found",
-        ));
+        return Err(Error::NoServerCert);
     }
 
     // Load private key
-    let private_key_item = with_buf_read(&tls_config.key, rustls_pemfile::read_one)?
-        .ok_or_else(|| io::Error::new(io::ErrorKind::Other, "No private key found"))?;
+    let private_key_item =
+        with_buf_read(&tls_config.key, rustls_pemfile::read_one)?.ok_or(Error::NoPrivateKey)?;
     let (Item::RSAKey(pkey) | Item::PKCS8Key(pkey) | Item::ECKey(pkey)) = private_key_item else {
-        return Err(io::Error::new(io::ErrorKind::Other, "No private key found"))
+        return Err(Error::NoPrivateKey);
```

Comment:
Should we have a distinction between *no* private key and an *incorrect* private key?